### PR TITLE
[W07H02] Added more track tests

### PR DESCRIPTION
**Pull request by HendrikHuebner**

### CODE REVIEW

File: w07h02/test/pgdp/security/TrackTest.java [hunk 1]
```diff
@@ -167,13 +167,28 @@ void assertSetRange(String type, boolean up, int start, int end, int amountOfPos
...
 
     @Test
     void setRangeCircle() {
         assertSetRange("down", true, 2, 1, 3, List.of(2, 0, 1));
+        assertSetRange("down", true, 3, 2, 5, List.of(3, 4, 5, 0, 1, 2));
+        assertSetRange("down", true, 1, 1, 3, List.of(1));
     }
 
+    @Test
+    void setRangeInvalid() {
```
Comment (julian-hartl) [inline w07h02/test/pgdp/security/TrackTest.java:182]: Could you remove this test? "die int-Parameter innerhalb der Array-Grenzen sind ( >= 0 und < posts.length)"

File: w07h02/test/pgdp/security/TrackTest.java [hunk 2]
```diff
@@ -167,13 +167,28 @@ void assertSetRange(String type, boolean up, int start, int end, int amountOfPos
...
+        assertSetRange("down", true, -2, 1, 3, List.of());
+        assertSetRange("down", true, 0, 5, 3, List.of());
+        assertSetRange("down", true, 5, 5, 3, List.of());
+        assertSetRange("down", true, 0, -1, 3, List.of());
+
+        assertSetRange("down", true, 0, 0, 0, List.of());
```
Comment (julian-hartl) [inline w07h02/test/pgdp/security/TrackTest.java:188]: You can keep this one though.

File: w07h02/test/pgdp/security/TrackTest.java [hunk 3]
```diff
@@ -221,11 +238,30 @@ void assertRemoveHazardAt(int start, int end, int capacity, List<Integer> expect
...
     @Test
     void removeHazardAtCircle() {
         assertRemoveHazardAt(3, 1, 4, List.of(3, 0, 1));
+        assertRemoveHazardAt(10, 5, 15, List.of(10, 11, 12, 13, 14, 0, 1, 2, 3, 4, 5));
+        assertRemoveHazardAt(3, 3, 5, List.of(3));
+    }
+
+    @Test
+    void hazardInvalid() {
+        assertRemoveHazardAt(-1, 3, 5, List.of());
```
Comment (julian-hartl) [inline w07h02/test/pgdp/security/TrackTest.java:254]: Same here.

File: w07h02/test/pgdp/security/TrackTest.java [hunk 4]
```diff
@@ -221,11 +238,30 @@ void assertRemoveHazardAt(int start, int end, int capacity, List<Integer> expect
...
+        assertRemoveHazardAt(1, -3, 5, List.of());
+        assertRemoveHazardAt(99, 3, 5, List.of());
+        assertRemoveHazardAt(1, 99, 5, List.of());
+
+        assertCreateHazardAt(-1, 3, 5, List.of());
+        assertCreateHazardAt(1, -3, 5, List.of());
+        assertCreateHazardAt(99, 3, 5, List.of());
+        assertCreateHazardAt(1, 99, 5, List.of());
+
+        assertCreateHazardAt(0, 0, 0, List.of());
```
Comment (julian-hartl) [inline w07h02/test/pgdp/security/TrackTest.java:264]: Except for this one.

File: w07h02/test/pgdp/security/TrackTest.java [hunk 5]
```diff
@@ -262,25 +298,40 @@ void assertRemoveLappedCarAt(int postAt, int capacity, List<Integer> expectedCal
...
     void removeLappedCarAtArrayBiggerThanThree() {
         assertRemoveLappedCarAt(0, 8, List.of(0, 1, 2, 3));
         assertRemoveLappedCarAt(7, 8, List.of(7, 0, 1, 2));
         assertRemoveLappedCarAt(5, 10, List.of(5, 6, 7, 8));
+        assertRemoveLappedCarAt(9, 10, List.of(9, 0, 1, 2));
+    }
+
+    @Test
+    void lappedCarInvalid() {
+        assertCreateLappedCarAt(-1, 5, List.of());
```
Comment (julian-hartl) [inline w07h02/test/pgdp/security/TrackTest.java:330]: Same here.

File: w07h02/test/pgdp/security/TrackTest.java [hunk 6]
```diff
@@ -262,25 +298,40 @@ void assertRemoveLappedCarAt(int postAt, int capacity, List<Integer> expectedCal
...
+        assertCreateLappedCarAt(99, 5, List.of());
+        assertRemoveLappedCarAt(-1, 5, List.of());
+        assertRemoveLappedCarAt(99, 5, List.of());
+        assertRemoveLappedCarAt(0, 0, List.of());
```
Comment (julian-hartl) [inline w07h02/test/pgdp/security/TrackTest.java:334]: Except for this one.

File: w07h02/test/pgdp/security/TrackTest.java (hunk 1 above)
Comment (HendrikHuebner) [reply w07h02/test/pgdp/security/TrackTest.java:182]: ok

**Comment by julian-hartl** [comment]
@nilsreichardt @HendrikHuebner ready to be merged.

**Comment by tdphucvn** [comment]
lgtm

**Review by MaximilianAnzinger — COMMENTED** [review]
🐧

**Pull request merged**